include/librealsense2/dds/dds-device-server.cpp

```diff
@@ -86,6 +90,171 @@ class dds_device_server::dds_stream_server
     image_header _image_header;
 };
 
+//------------------------------------------------------------------------------------------------------------//
+// DDS notifications server is in charge of publishing notifications and info like streams, profiles, events
+//------------------------------------------------------------------------------------------------------------//
+// We want to know when readers join our topic
+class dds_device_server::dds_notifications_server
+{
+public:
+    // We want to know when readers join our topic
+    class dds_notifications_client_listener : public eprosima::fastdds::dds::DataWriterListener
+    {
+    public:
+        dds_notifications_client_listener( dds_notifications_server * owner )
+            : eprosima::fastdds::dds::DataWriterListener()
+            , _owner( owner )
+        {
+        }
+
+        void on_publication_matched(
+            eprosima::fastdds::dds::DataWriter * writer,
+            const eprosima::fastdds::dds::PublicationMatchedStatus & info ) override
+        {
+            if( info.current_count_change == 1 )
+            {
+                LOG_DEBUG( "DataReader " << writer->guid() << " for topic: " << writer->get_topic()->get_name() << " discovered" );
```

Comment:
Won't the participant also get this and put out a message?
If so: do we really need another listener? Can't we add a listener to the participant and get the notification from there?

Reply:
Good question, I will read the docs on the difference between this 2 callbacks and we can decide together

Reply:
OK,

I read about it in `FastDDS` docs and they say:
```
events are always notified to the most specific Entity Listener that can handle the event, callbacks that DomainParticipantListener inherits from other Listeners will only be called if no other Entity was able to handle the event
```

It looks like it's possible to do all the work inside the participant listener but it has some disadvantages:
It will call all the callbacks and they will have to decide if the reader match their writer (topic name / QOS), and it will happen for each topic we have, meaning irrelevant CPU time consumption and locks even when the new reader is not for the topic we want to read/write.
I also spoke with @OhadMeir and we think a specific listener is more efficient here.

Reply:
I understand.

Not sure I like it.

We have callbacks on the participant, but they'll get called if nobody else takes them, and yet from the "outside", a user of our classes doesn't know how we implemented the internals. I.e., the more we add inside the less useful the participant becomes.

Reply:
It also makes the code VERY hard to follow.

Reply:
I agree it's more elegant to use the participant callback, but I think we better make this code section more efficient.
Maybe we can think of a refactor for it later on?
The main flow will stay the same probably.